File: marcenaria-java-activej/src/main/java/github/gmess/models/geometrias/Paralelepipedo.java
package github.gmess.models.geometrias;

import com.dslplatform.json.CompiledJson;
import com.dslplatform.json.JsonAttribute;
import github.gmess.utils.NumberUtils;

@CompiledJson(name = "paralelepipedo")
public record Paralelepipedo(
    @JsonAttribute(name = "estrutura") 
    String estruturaParalelepipedo,
    String altura,
    String comprimento,
    String largura
  ) implements Geometria {
  @Override
  public final String getGeometria() {
    return "cilindro";
  }

  @Override
  public final String getEstrutura() {
    return this.estruturaParalelepipedo;
  }

  @Override
  public final double getArea() {
    final double altura = NumberUtils.centimetersToDouble(this.largura);
    final double comprimento = NumberUtils.centimetersToDouble(this.comprimento);
    final double largura = NumberUtils.centimetersToDouble(this.altura);

    return 2 * (comprimento * largura + comprimento * altura + largura * altura);
  };
}

File: MarcenariaDotNet/Dtos/Geometrias/Paralelepipedo.cs
using System.Text.Json.Serialization;
using MarcenariaDotNet.Utils;

namespace MarcenariaDotNet.Dtos.Geometrias;

public struct Paralelepipedo : Geometria
{
  public String Estrutura { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Altura { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Comprimento { get; init; }
  [JsonConverter(typeof(CentimeterConverter))]
  public double Largura { get; init; }

  double Geometria.GetArea()
  {
     return 2 * Comprimento * Largura + Comprimento * Altura + Largura * Altura;
  }

  string Geometria.GetEstrutura()
  {
    return Estrutura;
  }

  string Geometria.GetGeometria()
  {
    return "paralelepipedo";
  }
}
